book/translation.rst

```diff
@@ -428,6 +428,14 @@ via the ``request`` object::
 
         $request->setLocale('en_US');
     }
+    
+    
+.. note::
+
+    Setting the locale using ``$request->setLocale()`` won't affect rendering
+    in the same action as the translator reads the request locale during the
+    kernel.request event, so changing it here would be too late. To manually
```

Comment:
This is a really long sentence, let's break it up. Something among the lines of:

````rst
Setting the locale using the ``$request->setLocale()`` method does not affect
rendering in the same action. The translator locale is set during the ``kernel.request``
event. Either change the locale before the listener is called or use the ``setLocale()``
method of the ``translator`` service.